contracts/NodeVersionBeacon.cdc

```diff
@@ -90,14 +90,18 @@ pub contract ExecutionNodeVersionBeacon {
     }
 
     /// Event emitted any time a change is made to versionTable
-    pub event ExecutionNodeVersionTableAddition(height: UInt64, version: Semver)
-    pub event ExecutionNodeVersionTableDeletion(height: UInt64, version: Semver)
+    // pub event NodeVersionTableAddition(height: UInt64, version: Semver)
+    // pub event NodeVersionTableDeletion(height: UInt64, version: Semver)
+    pub event NodeVersionTableUpdated(
+        versionTable: {UInt64: Semver},
```

Comment:
We need this table as an array of pairs, ordered. Something like
`[ { height, semver}, { height, semver }]`

Reply:
Something like this would make sense?
```
    pub event NodeVersionTableUpdated(
        versionTable: [{UInt64: Semver}],
        sequence: UInt64
    )
```

Since we only have arrays and dictionaries I can't think of other way of storing that info (each array position will be a dictionary with a single entry)

Reply:
Some alternatives to an array of dictionaries would be an array of structs defining a version boundary, something like:

```
pub struct VersionBoundary {
    pub let blockHeight: UInt64
    pub let version: Semver
}
```

so the emitted event type would be `[ VersionBoundary, ... ]`

Or, you could emit an array of arrays, each sub-array of length 2 - `[AnyStruct; 2]`. The emitted value would look  like `[ [ height, semver ], ... ]`

Reply:
Thanks Gio! @m4ksio do you have any preference?

Reply:
I think the version proposed by @sisyphusSmiling would be great - an array of structs is most concrete data type.